pkg/apis/hco/v1alpha1/hyperconverged_resources.go

```diff
@@ -19,23 +21,27 @@ func (r *HyperConverged) getNamespace(defaultNamespace string, opts []string) st
 	return defaultNamespace
 }
 
-func (r *HyperConverged) getLabels() map[string]string {
+func (r *HyperConverged) getLabels(component hcoutil.AppComponent) map[string]string {
 	hcoName := hcoutil.HyperConvergedName
 
 	if r.Name != "" {
 		hcoName = r.Name
 	}
 
 	return map[string]string{
-		hcoutil.AppLabel: hcoName,
+		hcoutil.AppLabel:          hcoName,
+		hcoutil.AppManagedByLabel: hcoutil.OperatorName,
+		hcoutil.AppVersionLabel:   os.Getenv(hcoutil.HcoKvIoVersionName),
```

Comment:
Correcting myself as per @yuvalturg comment, there aren't syscalls involved in reading the Environment variable - these are served by the runtime from memory, and hence the cost of `Getenv()` multiple times is reasonable. Nevertheless this might be somewhat cleaner to read them once, store, and serve from a state variable.

Reply:
fixed.